src/Dialect/ONNX/Transforms/ConstProp.cpp

```diff
@@ -287,6 +372,30 @@ constexpr auto subCombiner(Type elemType) {
   return elementwiseBinaryOpCombiner<ONNXSubOp>(elemType);
 }
 
+/// Precondition: min values must be less than max values if both exist.
+bool satisfiesMinLessThanMaxRequirement(ShapedType type, Value min, Value max) {
```

Comment:
It looks like checking this precondition in the ClipOp's verifier is more appropriate since it applies to the non-constant case also.